src/api/Client.ts

```diff
@@ -25,14 +24,17 @@ export const isOfTypeNetwork = (value: string): value is Network => {
 
 export const changeClientNetwork = (network: Network) => {
   if (currentNetwork === network) return;
-
   currentNetwork = network;
-  const baseURL = getBaseURL(network);
 
-  createNewClient(baseURL);
+  client = network === "MAINNET" ? mainnetClient : testnetClient;
   resetQueries();
 };
 
 export const getClient = () => {
   return client;
 };
+
+export const getOtherClient = () => {
```

Comment:
maybe you could have parametrized client
```suggestion
export const getClient = (network = "MAINNET") => {
const client =  clients[network];
if (client) {
  return client;
} else {
  throw Error("client instance not found");
}
}
```